integration-test/push_test.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+/* eslint-env mocha */
+
+const assert = require('assert')
+const { describe, it, before } = require('mocha')
+
+const { getTestNodeId } = require('../test/util')
+const { MediachainNode: AlephNode } = require('../src/peer/node')
+const { concatNodeClient, concatNodePeerInfo } = require('./util')
+const { generatePublisherId } = require('../src/peer/identity')
+
+const seedObjects = [
+  {id: 'foo:1', foo: 'bar'},
+  {id: 'foo:2', foo: 'baz'}
+]
+
+function seedStatementsToAleph (alephNode: AlephNode): Promise<Array<string>> {
+  return Promise.all(
+    seedObjects.map(obj =>
+      alephNode.ingestSimpleStatement('scratch.test', obj, { refs: [obj.id] })
+    )
+  )
+}
+
+function seedUnauthorizedStatement (alephNode: AlephNode): Promise<string> {
+  const obj = {letMeIn: 'please'}
+  return alephNode.ingestSimpleStatement('members.only', obj, { refs: ['foo'] })
+}
+
+describe('Push', () => {
+  let alephNode
+  let alephPeerIdB58
+  let publisherId
+  let statementIds
+  let unauthorizedStatementId
+
+  before(() => generatePublisherId()
+    .then(_publisherId => { publisherId = _publisherId })
+    .then(() => getTestNodeId())
+    .then(nodeId => {
+      const peerId = nodeId
+      alephPeerIdB58 = peerId.toB58String()
+      alephNode = new AlephNode({peerId, publisherId})
+    })
+    .then(() => seedStatementsToAleph(alephNode))
+    .then(_statementIds => { statementIds = _statementIds })
+    .then(() => seedUnauthorizedStatement(alephNode))
+    .then(_stmtId => { unauthorizedStatementId = _stmtId })
+    .then(() => concatNodeClient())
+    .then(concat => concat.authorize(alephPeerIdB58, ['scratch.*']))
+    .catch(err => {
+      console.error('error during push setup:', err)
```

Comment:
ah, sorry bout that! I stuck that in when things were failing during setup so I could print a stack trace - need to remove it.  well spotted :)